test/analysis-tests.ts

```diff
@@ -65,14 +66,20 @@ describe('LLVM-mca tool definition', () => {
         opts.should.be.deep.equal(['-o', 'output.txt']);
     });
 
-    it('should split if disabledFilters is a string', () => {
-        const info = {
-            exe: null,
-            remote: true,
-            lang: 'analysis',
-            disabledFilters: 'labels,directives',
-        };
-        const a = new LLVMmcaTool(info, ce);
-        a.compiler.disabledFilters.should.be.deep.equal(['labels', 'directives']);
-    });
+    // CompilerInfo.disabledFilters is now typed as string[] which doesn't allow string
+    // Since CompilerInfo only a type (and not a class) I assume something in the
+    // BaseComiler/AnalsysTool/LLVMmcaTool was doing the splitting logic
+    //
+    // I see two  options here:
+    //   - Modify disableFilters type to be string|string[] and fix typing issue sother places
+    //   - Remove this test because it seems like the intent is to only support string[] (which is less complicated)
+    // it('should split if disabledFilters is a string', () => {
+    //     const info: Partial<CompilerInfo> = {
+    //         remote: true,
+    //         lang: 'analysis',
+    //         disabledFilters: 'labels,directives',
+    //     };
+    //     const a = new LLVMmcaTool(info as CompilerInfo, ce);
+    //     a.compiler.disabledFilters.should.be.deep.equal(['labels', 'directives']);
+    // });
```

Comment:
I see, that's interesting

I'm not too familar with this part of the codebase, but LLVMmcaTool appears to take a `ToolInfo` instead of a CompilerInfo. Maybe this has something to do with it?

https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/tooling/base-tool.ts#L54

Reply:
It is supposed to consume ToolInfo at the constructor, yes. The CompilerInfo gets passed later at runTool()
So I'm a bit confused about how this was a test. Would just delete, perhaps.

Reply:
I think there's two LLVMmcaTool classes - one is a Tool (and takes a ToolInfo constructor) and one is an Analsys compiler (the one here) wihch takes a CompilerInfo. See https://github.com/compiler-explorer/compiler-explorer/pull/4740 for how I fixed this in my code)